### DISCUSSION TITLE
Matrix Rain, Screen Transitions and Palette Explorations

### DISCUSSION THREAD

**Discussion by lllama** [Textualize/textual#1510] [Show and tell]
Replicating some of my posts from Discord.

Matrix Rain in a container:

https://user-images.githubusercontent.com/598411/211091047-a3918947-6015-473d-bc71-51ce5583ba6a.mp4

Animated transitions between screens:

https://user-images.githubusercontent.com/598411/211091332-b1b1cfd3-440b-4d94-a489-79eec52489c9.mp4

Palettes from [Lospec](https://lospec.com/)

https://user-images.githubusercontent.com/598411/211091513-04102e58-838d-486b-ae91-627c5c6e60c1.mp4

Password decoding:

https://user-images.githubusercontent.com/598411/211091769-fc633310-938c-496d-a68c-85b8aa0ae3aa.mp4

**Comment by lllama** [comment]
Flicker free screen transitions!

https://user-images.githubusercontent.com/598411/212348077-62448fbe-a8bf-4817-ad7a-1a68aa14414a.mp4

**Comment by lllama** [comment]
Fading transition 

https://user-images.githubusercontent.com/598411/212467341-16145ad0-1694-4d10-a977-3243c7c02ad0.MOV

**Comment by willmcgugan** [comment]
I have to tweet this...

**Comment by wolf** [comment]
I would love to see the code for this (especially Transitions).  Will you be sharing?

**Reply by lllama** [reply]
That's the plan. It's a little too hacky at the moment but I'm working on tidying it up.

**Comment by lllama** [comment]
Morphing between widgets with the same ID and fading other content

https://user-images.githubusercontent.com/598411/212940203-a533c9cb-b48d-4b66-b96b-dda879bb154a.mp4

**Comment by lllama** [comment]
https://user-images.githubusercontent.com/598411/213196018-cdfda882-0d86-42a9-bda9-43b5265dbd8a.mp4

Very, very rough code here: https://github.com/lllama/textual-transitions
